src/instructlab/rag/process_docs.py

```diff
@@ -0,0 +1,156 @@
+        logger.info(f"Found {len(knowledge_files)} knowledge files")
+        logger.info(f"{knowledge_files}")
+
+        process_docs_from_folder(temp_dir, output_dir)
+
+
+# Copied from sdg.utils.chunkers because that code is being refactored so we want to avoid importing anything from it.
+# TODO: Once the code base has settled down, we should make sure this code exists only in one place.
+def resolve_ocr_options() -> OcrOptions:
+    # First, attempt to use tesserocr
+    try:
+        ocr_options = TesseractOcrOptions()
+
+        _ = TesseractOcrModel(True, ocr_options)
+        return ocr_options
+    except ImportError:
+        # No tesserocr, so try something else
+        pass
+    try:
+        ocr_options = EasyOcrOptions()
+        # Keep easyocr models on the CPU instead of GPU
+        ocr_options.use_gpu = False
+        # triggers torch loading, import lazily
+
+        _ = EasyOcrModel(True, ocr_options)
+        return ocr_options
+    except ImportError:
+        # no easyocr either, so don't use any OCR
+        logger.error(
+            "Failed to load Tesseract and EasyOCR - disabling optical character recognition in PDF documents"
+        )
+        return None
+
+
+def process_docs_from_folder(input_dir, output_dir, docling_model_path=None):
+    """
+    Process user documents from a given `input_dir` folder to the given `output_dir` folder, using docling converters.
+    Latest version of docling schema is used (currently, v2).
+    """
+    logger.info(f"Processing {input_dir} to {output_dir}")
+
+    clear_directory(Path(output_dir))
+
+    source_files = _load_source_files(input_dir=input_dir)
+    logger.info(f"Transforming source files {[p.name for p in source_files]}")
+
+    pipeline_options = PdfPipelineOptions(
+        artifacts_path=docling_model_path,
+        do_ocr=False,
+    )
+    ocr_options = resolve_ocr_options()
+    if ocr_options is not None:
+        pipeline_options.do_ocr = True
+        pipeline_options.ocr_options = ocr_options
+
+    doc_converter = DocumentConverter(
+        format_options={
+            InputFormat.PDF: PdfFormatOption(pipeline_options=pipeline_options)
+        }
+    )
+    start_time = time.time()
+    conv_results = doc_converter.convert_all(
+        source_files,
+        raises_on_error=False,
+    )
+    _, _, failure_count = _export_documents(conv_results, output_dir=Path(output_dir))
+    end_time = time.time() - start_time
+    logger.info(f"Document conversion complete in {end_time:.2f} seconds.")
+
+    if failure_count > 0:
+        raise RuntimeError(
+            f"The example failed converting {failure_count} on {len(source_files)}."
+        )
+
+
+def _load_source_files(input_dir) -> list[Path]:
+    return [
+        Path(os.path.join(input_dir, f))
+        for f in os.listdir(input_dir)
+        if os.path.isfile(os.path.join(input_dir, f))
+    ]
+
+
+def _export_documents(
```

Comment:
In case anyone asks the same, this function was taken from the Docling [Batch conversion](https://ds4sd.github.io/docling/examples/batch_convert/) example